src/app/view/view.component.ts

```diff
@@ -39,14 +42,36 @@ export default class ViewComponent {
   private readonly _dialogService = inject(DialogService);
   private readonly _exportService = inject(ExportModalService);
   private readonly _analyticsService = inject(AnalyticsService);
+  private readonly _router = inject(Router);
 
   protected readonly heroPencilSquareMini = heroPencilSquareMini;
   protected readonly heroPlusMini = heroPlusMini;
   protected readonly heroArrowDownTrayMini = heroArrowDownTrayMini;
+  protected readonly heroArrowLeftMini = heroArrowLeftMini;
+
+  public readonly id = input.required<string>();
 
   protected readonly palette = this._paletteService.palette;
   protected readonly saving = signal(false);
 
+  private _hasUnsavedChanges = false;
+
+  public get hasUnsavedChanges(): boolean {
+    return this._hasUnsavedChanges;
+  }
```

Comment:
As already discussed, I unfortunately cannot compute `hasUnsavedChanges` automatically with a `computed` since the reference to the palette is still the same after changes. So a `computed` wouldn't get re-evaluated when changes happen.

Since I didn't use the property anywhere in the template it made no sense to convert this into a signal (using a signal is slightly more expensive than a normal variable, since Angular has to manage the whole signal dependency stuff).

But you gave me the idea, that I could use this property to disable the "Save" button while there are not changes in the palette. I added it in this commit: https://github.com/pawcoding/Rainbow-Palette/pull/61/commits/180a94eecae3f6b5f395eae74567c65b03b32fea